x/concentrated-liquidity/lp_test.go

```diff
@@ -855,16 +858,16 @@ func (s *KeeperTestSuite) TestUpdatePosition() {
 			expectedError:             false,
 		},
 		"update existing position with negative amount": {
-			poolId:         1,
-			ownerIndex:     0,
-			lowerTick:      DefaultLowerTick,
-			upperTick:      DefaultUpperTick,
-			joinTime:       DefaultJoinTime,
-			positionId:     DefaultPositionId,
-			liquidityDelta: DefaultLiquidityAmt.Neg(), // negative
+			poolId:          1,
+			ownerIndex:      0,
+			lowerTick:       DefaultLowerTick,
+			upperTick:       DefaultUpperTick,
+			joinTime:        DefaultJoinTime,
+			positionId:      DefaultPositionId,
+			liquidityDelta:  DefaultLiquidityAmt.Neg(), // negative
+			amount0Expected: DefaultAmt0Expected.Neg(),
```

Comment:
It would be great if we can have a test case that covers asset 0 being rounded as well (e.g. one where we need to subtract 1 from asset 0 instead of asset 1)

Reply:
The problem with this is that I'd have to recalculate a lot of the values by hand, including ticks, and liquidity between them corresponding to the given amounts.

Let me make an issue to come back to this separately once we're in internal review mode

Reply:
https://github.com/osmosis-labs/osmosis/issues/5023